### PR TITLE
SINGA-475 add SoftPlus operator

### PR DESCRIPTION
**Pull request by pinpom**

### CODE REVIEW

File: test/python/test_operation.py [hunk 1]
```diff
@@ -610,6 +610,17 @@ def test_Atanh_gpu(self):
...
         self.check_shape(dx.shape(), (3, 2))
 
+    def test_SoftPlus(self):
+        X=np.array([1.0,2.0,3.0,4.0,5.0,6.0]).reshape(3,2).astype(np.float32)
+        XT=np.log(np.exp(X) + 1)
+        x=tensor.from_numpy(X)
+        x.to_device(gpu_dev)
+
+        result=autograd.softplus(x)
+        dx=result.creator.backward(x.data)
```
Comment (nudles) [inline test/python/test_operation.py]: pls also test the correctness of the gradients.

Comment (pinpom) [reply test/python/test_operation.py]: I added gradient test

File: python/singa/autograd.py [hunk 2]
```diff
@@ -1828,3 +1828,27 @@ def backward(self, dy):
...
+    #f(x) = ln(exp(x) + 1)
+        if training:
+            self.input = x
+        x1 = singa.AddFloat(singa.Exp(x),1.0)
+        y = singa.Log(x1)
+        
+        return y
+​
+    def backward(self, dy):
+        dx = singa.AddFloat(singa.Exp(self.input),1.0)
```
Comment (joddiy) [inline python/singa/autograd.py]: please use the formula: dx = 1 / (1 + exp(-x)). Your implement here has to compute the exp(x) twice.

Comment (pinpom) [reply python/singa/autograd.py]: yes. backward function changed.

**Comment by joddiy** [comment]
ready for merge

**Comment by chrishkchris** [comment]
I have run the test in docker container using cloud K80. It is ready for merge.

**Pull request merged**